src/query/algorithms/triangles/Triangles.cpp

```diff
@@ -105,23 +100,24 @@ TriangleResult Triangles::countTriangles(map<long, unordered_set<long>> &central
 
     long triangleCount = 0;
     std::unordered_map<long, std::unordered_map<long, std::unordered_set<long>>> triangleTree;
+    // TODO(thevindu-w): Make centralstore undirected and prevent saving triangles in-memory
```

Comment:
Can we please fix this TODO as well within this PR?

Reply:
This is the memory optimization. It should be a separate PR as this PR is intended for auto-scaling.

Reply:
Ok, I see